addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRuleUnitTest.java

```diff
@@ -121,4 +129,225 @@ public void replaceMultipleElementsAndRemoveHeader() {
                         + "</comments>";
         assertThat(payload, is(expectedPayload));
     }
+
+    @Test
+    public void shouldScanOnlyIfRequestContentTypeIsXml() throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response =
+                                "<foo>root:*:0:0:System Administrator:/var/root:/bin/sh</foo>";
+                        return newFixedLengthResponse(
+                                NanoHTTPD.Response.Status.OK, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setHeader("Content-Type", "application/json");
+        msg.getRequestHeader().setMethod("POST");
+        rule.init(msg, parent);
+
+        // When
+        rule.scan();
+
+        // Then
+        assertThat(countMessagesSent, equalTo(0));
+    }
+
+    @Test
+    public void shouldSendThreeMessagesForLocalFileReflectionRule() throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response =
+                                "<foo>root:*:0:0:System Administrator:/var/root:/bin/sh</foo>";
+                        return newFixedLengthResponse(
+                                NanoHTTPD.Response.Status.OK, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setMethod("POST");
+        msg.getRequestHeader().setHeader("Content-Type", "application/xml");
+        rule.init(msg, parent);
+
+        // When
```

Comment:
The blanks leading `// When` and `// Then` throughout can be removed, the comments already delineate things.

Reply:
This has been removed now.